core/contractsdk/go/example/award_manage/main.go

```diff
@@ -0,0 +1,253 @@
+		return code.Error(err)
+	}
+
+	return code.OK([]byte(args.TotalSupply.String()))
+}
+
+func (am *awardManage) AddAward(ctx code.Context) code.Response {
+	caller := ctx.Caller()
+	if caller == "" {
+		return code.Error(utils.ErrMissingCaller)
+	}
+	masterBytes, err := ctx.GetObject([]byte(MASTERPRE))
+	if err != nil {
+		return code.Error(err)
+	}
+	master := string(masterBytes)
+
+	if master != caller {
+		return code.Error(utils.ErrPermissionDenied)
+	}
+
+	args := struct {
+		Amount *big.Float `json:"amount" gt:"0"`
+	}{}
+
+	if err := utils.Validate(ctx.Args(), &args); err != nil {
+		return code.Error(err)
+	}
+
+	totalSupplyByte, err := ctx.GetObject([]byte(TOTAL_SUPPLY))
+	if err != nil {
+		return code.Error(err)
+	}
+
+	totalSupply, _ := big.NewFloat(0).SetString(string(totalSupplyByte))
+	totalSupply = big.NewFloat(0).Add(totalSupply, args.Amount)
+
+	if err := ctx.PutObject([]byte(TOTAL_SUPPLY), []byte(totalSupply.String())); err != nil {
+		return code.Error(err)
+	}
+	return code.OK([]byte(totalSupply.String()))
+}
+
+func (am *awardManage) TotalSupply(ctx code.Context) code.Response {
+	if value, err := ctx.GetObject([]byte(TOTAL_SUPPLY)); err != nil {
+		return code.Error(err)
+	} else {
+		return code.OK(value)
+	}
+}
+
+func (am *awardManage) Balance(ctx code.Context) code.Response {
+	caller := ctx.Caller()
+	if caller == "" {
+		return code.Error(utils.ErrMissingCaller)
+	}
+	value, err := ctx.GetObject([]byte(BALANCEPRE + caller))
+	if err != nil {
+		return code.Error(err)
+	}
+	return code.OK(value)
+}
+
+func (am *awardManage) Allowance(ctx code.Context) code.Response {
+	args := struct {
+		From string `json:"from"`
+		To   string `json:"to"`
+	}{}
+	if err := utils.Validate(ctx.Args(), &args); err != nil {
+		return code.Error(err)
+	}
+
+	value, err := ctx.GetObject([]byte(ALLOWANCEPRE + args.From + "_" + args.To))
+	if err != nil {
+		return code.Error(err)
+	}
+	return code.OK(value)
+}
+
+func (am *awardManage) Transfer(ctx code.Context) code.Response {
+	caller := ctx.Caller()
+	from := caller
+	if from == "" {
+		return code.Error(utils.ErrMissingCaller)
+	}
+	args := struct {
+		To    string     `json:"to" required:"true"`
+		Token *big.Float `json:"token" required:"true"`
+	}{}
+
+	if err := utils.Validate(ctx.Args(), &args); err != nil {
+		return code.Error(err)
+	}
+	if from == args.To {
+		return code.Error(errors.New("can not transfer to yourself"))
+	}
+
+	fromBalanceByte, err := ctx.GetObject([]byte(BALANCEPRE + from))
+	if err != nil {
+		return code.Error(err)
+	}
+	fromBalance, _ := big.NewFloat(0).SetString(string(fromBalanceByte))
+
+	if fromBalance.Cmp(args.Token) < 0 {
+		return code.Error(errors.New("balance not enough"))
+	}
+
+	toBalanceByte, err := ctx.GetObject([]byte(BALANCEPRE + args.To))
+	if err != nil { // errors!=nil means account not found
+		toBalanceByte = []byte("0")
+	}
+	toBalance, _ := big.NewFloat(0).SetString(string(toBalanceByte))
```

Comment:
Is it possible that `big.NewFloat(0).SetString` will response with error?

Reply:
yeah, it may return false if it can not be parsed as an float point number. howerve,  we choose to believe on xchain. 
the desision stratege is

- if the string is from contract method arguements （ctx.args ）, it must be checked
- if the string is from contex, it is not neccessary to be checked

Reply:
by the way, all big.Float has been replaced with big.Int